crates/fuzzing/src/oracles/diff_wasmtime.rs

```diff
@@ -0,0 +1,184 @@
+struct WasmtimeInstance {
+    store: RefCell<Store<StoreLimits>>,
+    instance: Instance,
+}
+
+impl DiffInstance for WasmtimeInstance {
+    fn name(&self) -> &'static str {
+        "wasmtime"
+    }
+
+    fn evaluate(&mut self, function_name: &str, arguments: &[DiffValue]) -> Result<Vec<DiffValue>> {
+        let arguments: Vec<_> = arguments.iter().map(Val::from).collect();
+
+        let function = self
+            .instance
+            .get_func(self.store.borrow_mut().deref_mut(), function_name)
+            .expect("unable to access exported function");
+        let ty = function.ty(self.store.borrow().deref());
+        let mut results = vec![Val::I32(0); ty.results().len()];
+        function.call(
+            self.store.borrow_mut().deref_mut(),
+            &arguments,
+            &mut results,
+        )?;
+
+        let results = results.into_iter().map(Val::into).collect();
+        Ok(results)
+    }
+
+    fn is_hashable(&self) -> bool {
+        true
+    }
+
+    fn hash(&self, state: &mut std::collections::hash_map::DefaultHasher) -> Result<()> {
+        for e in self.instance.exports(self.store.borrow_mut().deref_mut()) {
+            match e.into_extern() {
+                Extern::Global(g) => {
+                    let val: DiffValue = g.get(self.store.borrow_mut().deref_mut()).into();
+                    val.hash(state)
+                }
+                Extern::Memory(m) => {
+                    let mut store = self.store.borrow_mut();
+                    let data = m.data(store.deref_mut());
+                    data.hash(state)
+                }
+                Extern::SharedMemory(m) => {
+                    let data = unsafe { slice::from_raw_parts(m.data() as *mut u8, m.data_size()) };
+                    data.hash(state)
+                }
+                Extern::Table(_) => {
+                    // TODO: it's unclear whether it is worth it to iterate
+                    // through the table and hash the values.
+                    todo!()
+                }
+                Extern::Func(_) => {
+                    // Note: no need to hash exported functions.
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+impl From<&DiffValue> for Val {
+    fn from(v: &DiffValue) -> Self {
+        match *v {
+            DiffValue::I32(n) => Val::I32(n),
+            DiffValue::I64(n) => Val::I64(n),
+            DiffValue::F32(n) => Val::F32(n),
+            DiffValue::F64(n) => Val::F64(n),
+            DiffValue::V128(n) => Val::V128(n),
+        }
+    }
+}
+
+impl Into<DiffValue> for Val {
+    fn into(self) -> DiffValue {
+        match self {
+            Val::I32(n) => DiffValue::I32(n),
+            Val::I64(n) => DiffValue::I64(n),
+            Val::F32(n) => DiffValue::F32(n),
+            Val::F64(n) => DiffValue::F64(n),
+            Val::V128(n) => DiffValue::V128(n),
+            Val::FuncRef(_) => unimplemented!(),
+            Val::ExternRef(_) => unimplemented!(),
```

Comment:
I think that this is worth implementing but all that we can really say about these is "null" or "non-null" so I think we could have `DiffValue::ExternRef { is_null: bool }` perhaps

Reply:
The single-instruction modules that this PR uses don't exercise reference types but I agree that in the future, e.g., with `wasm-smith`-generated modules, this is probably what we want to do.